testing/python/metafunc.py

```diff
@@ -1780,3 +1780,40 @@ def test_foo(a):
         )
         result = testdir.runpytest()
         result.assert_outcomes(passed=1)
+
+    def test_parametrize_explicit_parameters_func(self, testdir):
+        testdir.makepyfile(
+            """
+            import pytest
+
+
+            @pytest.fixture
+            def fixture(arg):
+                return arg
+
+            @pytest.mark.parametrize("arg", ["baz"])
+            def test_without_arg(fixture):
+                assert "baz" == fixture
+        """
+        )
+        result = testdir.runpytest()
+        result.assert_outcomes(error=1)
```

Comment:
We should also check the actual message shown to the user. For that you can use `result.stdout.fnmatch_lines`:

```python
result.stdout.fnmatch_lines([
    '*In function "test_without_arg"*',    
    '*Parameter "arg" should be declared explicitly via indirect*',
    '*or in function itself*',
])
```

(same for the other test)

Reply:
Done! The interesting thing is that it seems like proposed functionality in general causes other test to fail, e.g.  `test_usefixtures_seen_in_generate_tests` in the same file (`testing/python/metafunc.py`). Any ideas how do deal with that? Would be really grateful

Reply:
Oh that test should be declaring `fix2`, which is exactly what the functionality now tells about.

Please just update the test:

```diff
diff --git a/testing/python/collect.py b/testing/python/collect.py
index 30f9841b5..cf530bf0f 100644
--- a/testing/python/collect.py
+++ b/testing/python/collect.py
@@ -463,7 +463,7 @@ class TestFunction:
                return '3'

             @pytest.mark.parametrize('fix2', ['2'])
-            def test_it(fix1):
+            def test_it(fix1, fix2):
                assert fix1 == '21'
                assert not fix3_instantiated
         """
```

Reply:
Sure! I've updated that one in the last commit